File: InteractiveLive/Android/VideoOne/component/LoginKit/src/main/java/com/vertcdemo/login/utils/AgreementManager.java
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0

package com.vertcdemo.login.utils;

import androidx.annotation.Keep;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

@SuppressWarnings("unused")
@Keep
public class AgreementManager {
    public interface ResultCallback {
        void onResult(boolean result);
    }

    public static <T extends AppCompatActivity & ResultCallback> void check(@NonNull T activity) {
        activity.onResult(true);
    }
}

File: Client/Android/component/LoginKit/src/main/kotlin/com/vertcdemo/login/utils/AgreementManager.kt
// Copyright (c) 2023 BytePlus Pte. Ltd.
// SPDX-License-Identifier: Apache-2.0
package com.vertcdemo.login.utils

import android.content.Context
import androidx.appcompat.app.AppCompatActivity

object AgreementManager {
    fun <T> check(activity: T) where T : AppCompatActivity, T : ResultCallback =
        activity.onResult(true)

    fun getAgreementStatus(context: Context?): Boolean = true

    interface ResultCallback {
        fun onResult(result: Boolean)
    }
}
